pkg/commands/test/test.go

```diff
@@ -229,41 +244,64 @@ func makeQuery(rule string) string {
 }
 
 func processData(ctx context.Context, input interface{}, compiler *ast.Compiler) (CheckResult, error) {
-	// collect warnings
-	var warnings []error
-	for _, rule := range getRules(ctx, WarnQ, compiler) {
-		warns, err := runQuery(ctx, makeQuery(rule), input, compiler)
-		if err != nil {
-			return CheckResult{}, err
-		}
 
-		warnings = append(warnings, warns...)
+	warnings, err := runRules(ctx, input, WarnQ, compiler)
+	if err != nil {
+		return CheckResult{}, err
+	}
+
+	failures, err := runRules(ctx, input, DenyQ, compiler)
+	if err != nil {
+		return CheckResult{}, err
+	}
+
+	result := CheckResult{
+		Warnings: warnings,
+		Failures: failures,
 	}
 
-	// collect failures
-	var failures []error
-	for _, r := range getRules(ctx, DenyQ, compiler) {
-		fails, err := runQuery(ctx, makeQuery(r), input, compiler)
+	return result, nil
+}
+
+func runRules(ctx context.Context, input interface{}, regex *regexp.Regexp, compiler *ast.Compiler) ([]error, error) {
+	var totalErrors []error
+	var errors []error
+	var err error
+
+	rules := getRules(ctx, regex, compiler)
+	for _, rule := range rules {
+		_, multiple := input.([]interface{})
```

Comment:
what about changing this with type checking switch-case like:

		switch input.(type) {
		case []interface{}:
			errors, err = runMultipleQueries(ctx, makeQuery(rule), input, compiler)
		case interface{}:
			errors, err = runQuery(ctx, makeQuery(rule), input, compiler)
		}

Reply:
I think this makes sense. All set!